### PR TITLE
Bench update exclude apps from updating

### PR DESCRIPTION
**Pull request by shridarpatil**
https://github.com/frappe/bench/issues/379

Commands: `bench exclude-app app-name`
Commands: `bench update --exclude-apps`

### CODE REVIEW

File: bench/app.py [hunk 1]
```diff
@@ -48,6 +48,29 @@ def write_appstxt(apps, bench_path='.'):
...
+	except IOError:
+		return []
+
+def add_to_excluded_appstxt(app, bench_path='.'):
+	apps = get_excluded_apps(bench_path=bench_path)
+	if app not in apps:
+		apps.append(app)
+		return write_excluded_appstxt(apps, bench_path=bench_path)
+
+def write_excluded_appstxt(apps, bench_path='.'):
```
Comment (saurabh6790) [inline bench/app.py]: should be `_txt`

File: bench/app.py [hunk 2]
```diff
@@ -48,6 +48,29 @@ def write_appstxt(apps, bench_path='.'):
...
+	with open(os.path.join(bench_path, 'sites', 'excluded_apps.txt'), 'w') as f:
+		return f.write('\n'.join(apps))
+
+def remove_from_exclided_appsstxt(app, bench_path='.'):
```
Comment (saurabh6790) [inline bench/app.py]: check spelling

File: bench/app.py [hunk 3]
```diff
@@ -48,6 +48,29 @@ def write_appstxt(apps, bench_path='.'):
...
+	apps = get_apps(bench_path=bench_path)
```
Comment (saurabh6790) [inline bench/app.py]: should be `get_excluded_apps` not `get_apps`

File: bench/app.py [hunk 4]
```diff
@@ -132,13 +155,17 @@ def remove_app(app, bench_path='.'):
 		restart_supervisor_processes(bench_path=bench_path)
 
 
-def pull_all_apps(bench_path='.', reset=False):
+def pull_all_apps(bench_path='.', reset=False, exclude=True):
 	'''Check all apps if there no local changes, pull'''
 	rebase = '--rebase' if get_config(bench_path).get('rebase_on_pull') else ''
-	
+	print("Pull all apps {}".format(exclude))
```
Comment (saurabh6790) [inline bench/app.py]: Why?

Comment (saurabh6790) [inline bench/app.py]: Why?

File: bench/commands/__init__.py [hunk 5]
```diff
@@ -70,4 +70,10 @@ def bench_command(bench_path='.'):
 bench_command.add_command(remote_urls)
 
 from bench.commands.install import install
-bench_command.add_command(install)
\ No newline at end of file
+bench_command.add_command(install)
+
+from bench.commands.exclude_update import exclude_update
```
Comment (saurabh6790) [inline bench/commands/__init__.py]: Import exclude_update and include_update like 
`from bench.commands.exclude_update import exclude_update, include_update`

Comment (saurabh6790) [inline bench/commands/__init__.py]: Why two imports?

do like `from bench.commands.exclude_update import exclude_update, include_update`

File: bench/commands/__init__.py [hunk 6]
```diff
@@ -70,4 +70,10 @@ def bench_command(bench_path='.'):
...
+bench_command.add_command(exclude_update)
+
+from bench.commands.exclude_update import include_update
+bench_command.add_command(include_update)
```
Comment (saurabh6790) [inline bench/commands/__init__.py]: Also name rename the methods to `include_app_for_update` and `exclude_app_for_update`

**Comment by shridarpatil** [comment]
@saurabh6790 Will make the changes

File: bench/app.py [hunk 7]
```diff
@@ -48,6 +48,29 @@ def write_appstxt(apps, bench_path='.'):
 	with open(os.path.join(bench_path, 'sites', 'apps.txt'), 'w') as f:
 		return f.write('\n'.join(apps))
 
+def get_excluded_apps(bench_path='.'):
+	try:
+		with open(os.path.join(bench_path, 'sites', 'excluded_apps.txt')) as f:
```
[the remark this answers is not in the document]
Comment (shridarpatil) [reply bench/app.py]: Here I am getting the list of excluded apps. So I think  `get_excluded_apps` is more specific then just `exclude_apps_for_update`

**Comment by saurabh6790** [comment]
@shridarpatil 👍 

Please rebase it!

**Pull request merged**